docs/reference/language/context.md

```diff
@@ -25,38 +25,180 @@ in some places that are also aware of their location in the document:
+
+When a property is changed, the response to the query changes accordingly: 
 
 ```example
-#let value = context text.lang
-#value
+#set text(lang: "en")
+#context text.lang
 
 #set text(lang: "de")
-#value
+#context text.lang
+```
+
+The output of a `#context ...` call is static in the form of opaque 
+[`content`]. Write access to context output is prohibited, as it would 
+often result in invalid code: If the context changes between read and 
+write, overwriting a property would cause an inconsistent system state. 
+In fact, context-dependent property fields are immutable constants even 
+within the context itself:
+
+```example
+#set text(lang: "en")
+#context [
+  call 1: #text.lang \
+
+  #set text(lang: "fr")
+  call 2: #text.lang
+]
+```
+
+Both calls have the same output 'en', because `text.lang` is assigned
+upon entry in the context and remains constant until the end of its scope 
+(the closing `]`). It does not "see" the `#set text(lang: "fr")` before 
+call 2. Compare this to the previous example: there we got two different 
+results because we created two different contexts.
+
+However, immutability only applies to the property fields themselves. 
+Content creation instructions within a context _do_ see the effect of 
+the set rule. Consider the same example with font size:
+
+```example
+#set text(size: 50pt)
+#context [
+  call 1: #text.size \
+
+  #set text(size: 25pt)
+  call 2: #text.size
+]
+```
+
+Call 2 still outputs '50pt', because `text.size` is a constant.
+However, this output is printed in '25pt' font, as specified by the set
+rule before the call. This illustrates the importance of picking the 
+right insertion point for a context to get access to precisely the right 
+styles. If you need access to updated property fields after a set rule, 
+you can use nested contexts:
+
+```example
+#set text(lang: "en")
+#context [
+  call 1: #text.lang \
+
+  #set text(lang: "fr")
+  call 2: #context text.lang
+]
+```
+
+All of the above applies to `show` rules analogously, for example:
+
+```example
+#let template(body) = {
+  set text(size: 25pt)
+  body
+}
+
+#set text(size: 50pt)
+#context [
+  call 1: #text.size \
+
+  #show: template
+  call 2: #text.size \
+  call 3: #context text.size
+]
+```
+
+## Controlling content creation within a context
+
+The main purpose of retrieving the current values of properties is, 
+of course, to use them in the calculation of derived properties, 
+instead of setting those properties manually. For example, you can
+double the font size like this:
+
+```example
+#context {
+  // the context allows you to
+  // retrieve the current text.size
+  set text(size: text.size * 200%)
+  [large text \ ] 
+}
+original size
+```
 
-#set text(lang: "fr")
-#value
+Since set rules are only active until the end of the enclosing scope, 
+'original size' is printed with the original font size.
+The above example is equivalent to 
+
+```example
+#{
+  set text(size: 2em)
+  [large text \ ]
+}
+original size
```

Comment:
The start of a phrase in example should be capitalized. E.g. "Original size" or "Call 1" (bad example because call is not the correct term, but just an example).